src/calendar.js

```diff
@@ -42,16 +43,17 @@ export default class Calendar extends Component {
 
   render() {
     const m = this.props.moment;
-    const d = m.date();
-    const d1 = m.clone().subtract(1, 'month').endOf('month').date();
-    const d2 = m.clone().date(1).day();
-    const d3 = m.clone().endOf('month').date();
-    const days = [].concat(
-      range(d1 - d2 + 1, d1 + 1),
-      range(1, d3 + 1),
-      range(1, 42 - d3 - d2 + 1)
-    );
-    const weeks = ['Sun', 'Mon', 'Tue', 'Wed', 'Thu', 'Fri', 'Sat'];
+    const date = m.date();
+
+    const monthStartWeekDay = m.clone().startOf('month').format('e');
+    const monthEndWeekDay = m.clone().endOf('month').format('e');
+    const dateStart = monthStartWeekDay === 0 ? m.clone().startOf('month') : m.clone().startOf('month').subtract(monthStartWeekDay, 'days');
+    const dateEnd = monthEndWeekDay === 6 ? m.clone().endOf('month') : m.clone().endOf('month').add(6-monthEndWeekDay, 'days');
+    const daysRange = moment.range(dateStart, dateEnd);
```

Comment:
**NOTE:** I used your locale changes in another project, but there I'd already split up all the logic into logic / presentational components, and only wanted the _entry point_ to have a dependency on `moment`. So I kept your locale stuff, and used the "old" `lodash.range` way 😢 

But for this self-contained single-component structure, it does make much more sense to let moment-range do the date-math 👍